Chocolate_Vending_Machine.java

```diff
@@ -0,0 +1,240 @@
+		}
+		return true;
+	}
+
+	public ArrayList<Coin> returnChange(ArrayList<Integer> money, int s_no, int qty)
+			throws NoSufficientChangeException {
+		ArrayList<Coin> changes = new ArrayList<>();
+		int coindeduct[] = new int[4];
+		int tot_Money = 0;
+		for (int i = 0; i < money.size() - 1; i++) {
+			tot_Money += money.get(i);
+			switch (money.get(i)) {
+			case 10:
+				addCoin(Coin.ten, 1);
+				break;
+			case 5:
+				addCoin(Coin.five, 1);
+				break;
+			case 2:
+				addCoin(Coin.two, 1);
+				break;
+			case 1:
+				addCoin(Coin.one, 1);
+				break;
+			}
+		}
+		Item items_in[] = Item.values();
+		int actualPrice = items_in[s_no - 1].getPrice(items_in[s_no - 1]) * qty;
+		int diffMoney = tot_Money - actualPrice;
+
+		if (diffMoney > 0) {
+			while (diffMoney > 0) {
+				if (diffMoney >= Coin.ten.getDenomination() && cash.get(Coin.ten) >= coindeduct[0]) {
+					diffMoney -= Coin.ten.getDenomination();
+					coindeduct[0]++;
+					changes.add(Coin.ten);
+				} else if (diffMoney >= Coin.five.getDenomination() && cash.get(Coin.five) >= coindeduct[1]) {
+					diffMoney -= Coin.five.getDenomination();
+					coindeduct[1]++;
+					changes.add(Coin.five);
+				} else if (diffMoney >= Coin.two.getDenomination() && cash.get(Coin.two) >= coindeduct[2]) {
+					diffMoney -= Coin.two.getDenomination();
+					coindeduct[2]++;
+					changes.add(Coin.two);
+				} else if (diffMoney >= Coin.one.getDenomination() && cash.get(Coin.two) >= coindeduct[3]) {
+					diffMoney -= Coin.one.getDenomination();
+					coindeduct[3]++;
+					changes.add(Coin.one);
+				} else {
+					throw new NoSufficientChangeException("Machine does not have sufficient change");
+				}
+			}
+		}
+		return changes;
+	}
+
+	private void updateCashInventory(ArrayList<Coin> changes) {
+		for (Coin c : changes) {
+			removeCoin(c, 1);
+		}
+	}
+
+	private void updateItemInventory(int s_no, int qty) {
+		Item items_in[] = Item.values();
+		removeItem(items_in[s_no - 1], qty);
+
+	}
+
+	public void collectMoney_And_UpdateInventories(ArrayList<Coin> changes, int s_no, int qty) {
+		Item items_in[] = Item.values();
+		System.out.println("Collect your money!! ");
+		if (changes.size() > 0) {
+			for (Coin c : changes) {
+				System.out.print(c.getDenomination() + " ");
+			}
+			System.out.println();
+		} else {
+			System.out.println("No return change left");
+		}
+		System.out.println("Your order ");
+		System.out.println("Item                      Quantity");
+		System.out.println(items_in[s_no - 1].getName(items_in[s_no - 1]) + "                	" + qty);
+		System.out.println();
+		System.out.println("Please Visit Again");
+		updateCashInventory(changes);
+		updateItemInventory(s_no, qty);
+	}
+
+	public void printInventoryStatus() {
+		Item items_in[] = Item.values();
+		for (Item i : items_in) {
+			System.out.println(i.getName(i) + "     " + items.get(i));
+		}
+	}
+
+	public static void main(String[] args) {

```

Comment:
main function of a vending machine should be some kind of infinite event loop waiting on user input